File: My Project/AndroidProject/AnimationDemo/app/src/main/java/com/theannguyen/animationdemo/MainActivity.java
package com.theannguyen.animationdemo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private TextView txtView,txtView2;
    private boolean  isHelloWorldShowing = true;
    int rot = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        txtView = (TextView) findViewById(R.id.txtView);
        txtView2 = (TextView) findViewById(R.id.txtView2);
        txtView2.setAlpha(0);
        txtView2.setX(-600);
        final float xText = txtView.getX();
        txtView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                if(isHelloWorldShowing){
//                    txtView.animate().alpha(0).setDuration(2000);
//                    txtView2.animate().alpha(1).setDuration(2000);
//                    isHelloWorldShowing = false;
//                }else if(!isHelloWorldShowing){
//                    txtView.animate().alpha(1).setDuration(2000);
//                    txtView2.animate().alpha(0).setDuration(2000);
//                    isHelloWorldShowing = true;
//                }

                //txtView.animate().rotationXBy(360).setDuration(2000);
                //txtView.animate().rotationBy(360).setDuration(1000);
                //txtView.animate().translationXBy(200).setDuration(1000);
                txtView2.animate().translationX(xText).alpha(1).rotationY(360).setDuration(3000);
                txtView.setAlpha(0);
            }
        });

    }
}

File: Android_Kotlin/AsyncTask/app/src/main/java/com/theannguyen/asynctask/MainActivity.kt
package com.theannguyen.asynctask

import android.os.AsyncTask
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button.setOnClickListener {
            Task().execute()
        }

    }
    inner class Task: AsyncTask<Void, String, String>() {
        override fun doInBackground(vararg params: Void?): String {

            for (sv in 1..3)
                publishProgress("publishProgress $sv \n")

            return "doInBackground"
        }

        override fun onPreExecute() {
            super.onPreExecute()
            textView.text = "onPreExecute \n"
        }

        override fun onPostExecute(result: String?) {
            super.onPostExecute(result)
            textView.append(result)
        }

        override fun onProgressUpdate(vararg values: String?) {
            super.onProgressUpdate(*values)
            textView.append(values[0])
        }

    }

}
